include/eve/detail/function/simd/x86/slide_left.hpp

```diff
@@ -0,0 +1,156 @@
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/implementation.hpp>
+#include <eve/detail/function/simd/x86/specifics.hpp>
+#include <eve/conditional.hpp>
+
+namespace eve::detail
+{
+  template<real_scalar_value T, typename N, x86_abi ABI, std::ptrdiff_t Shift>
+  EVE_FORCEINLINE wide<T,N,ABI> slide_left_ ( EVE_SUPPORTS(sse2_)
+                                            , wide<T,N,ABI> v, index_t<Shift>
+                                            ) noexcept
+  requires(Shift <= N::value)
+  {
+          if constexpr(Shift == 0)        return v;
+    else  if constexpr(Shift == N::value) return wide<T,N,ABI>{0};
+    else
+    {
+      if constexpr( std::same_as<ABI,x86_128_>)
+      {
+        constexpr auto shift = Shift*sizeof(T);
+        using i_t = as_integer_t<wide<T,N,ABI>, unsigned>;
+
+        auto const b  = bit_cast(v, as_<i_t>());
+        auto result = bit_cast(i_t(_mm_bsrli_si128( b, shift)), as(v));
+
+        // Mask noises from smaller sized registers
+        if constexpr(N::value < expected_cardinal_v<T,x86_128_>)
+        {
+          result &= keep_first(N::value-Shift).mask(as(result)).bits();
+        }
+
+        return result;
+      }
+      else if constexpr( std::same_as<ABI,x86_256_>)
+      {
+        if constexpr( current_api >= avx2)
+        {
+          constexpr auto offset = Shift * sizeof(T);
+          __m256i bv = _mm256_permute2x128_si256(v, v, 0x83);
+                if constexpr(offset == 16)  return bv;
+          else  if constexpr(offset <  16)  return _mm256_alignr_epi8(bv, v, offset);
+          else                              return _mm256_bsrli_epi128(bv, offset - 16);
+        }
+        else if constexpr(sizeof(T) == 8)
+        {
+          // Three optimal shuffles for sliding left by 1->3 - The last one will shock you!
+          using f_t = as_floating_point_t<wide<T,N,ABI>>;
+          auto const w = bit_cast(v, as_<f_t>{});
+
+          if constexpr(Shift == 1)
+          {
+            return bit_cast(f_t(_mm256_shuffle_pd(w,_mm256_permute2f128_pd(w,w,0x81),0x5)), as(v));
+          }
+          else if constexpr(Shift == 2)
+          {
+            return bit_cast(f_t(_mm256_permute2f128_pd(w, w, 0x81)), as(v));
+          }
+          else if constexpr(Shift == 3)
+          {
+            auto const s0 = _mm256_permute2f128_pd(w,w,0x81);
+            return bit_cast(f_t(_mm256_shuffle_pd(s0,f_t(0),1)), as(v));
+          }
+        }
+        else if constexpr(sizeof(T) == 4)
+        {
+          // Seven optimal shuffles for sliding left by 1->7 doctors want you to not know about
+          using f_t = as_floating_point_t<wide<T,N,ABI>>;
+          auto const w = bit_cast(v, as_<f_t>{});
+
+          if constexpr(Shift == 1)
```

Comment:
`_mm256_shuffle_ps`([a b c d e f g h], [0 0 0 0 a b c d]) for the < 4
`_mm256_shuffle_ps`([a b c d e f g h], [e f g k 0 0 0 0] ) for > 4